rpcserver.go

```diff
@@ -3251,12 +3251,17 @@ func handleSendRawTransaction(s *rpcServer, cmd interface{}, closeChan <-chan st
 		}
 	}
 
+	if len(acceptedTxs) == 0 || !acceptedTxs[0].Tx.Hash().IsEqual(tx.Hash()) {
+		s.server.txMemPool.RemoveTransaction(tx, true)
+		return nil, btcjson.ErrRPCInternal
+	}
+
 	s.server.AnnounceNewTransactions(acceptedTxs)
 
 	// Keep track of all the sendrawtransaction request txns so that they
 	// can be rebroadcast if they don't make their way into a block.
-	iv := wire.NewInvVect(wire.InvTypeTx, tx.Hash())
-	s.server.AddRebroadcastInventory(iv, tx)
+	iv := wire.NewInvVect(wire.InvTypeTx, acceptedTxs[0].Tx.Hash())
```

Comment:
No reason to change this to another array lookup.  The transaction hash is the same.

If you want them to be the same, that is acceptable, but please only do the lookup once.  i.e
```Go
	txD := acceptedTxs[0]
	iv := wire.NewInvVect(wire.InvTypeTx, txD.Tx.Hash())
	s.server.AddRebroadcastInventory(iv, txD)
```